### ISSUE TITLE
Test failed: JIT\Regression\CLR-x86-JIT\V1-M10\b04574\b04574\b04574.cmd

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/runtime#39681]
Windows x64

with:
```
set COMPlus_TieredCompilation=0
set COMPlus_GCStress=0xC
set COMPlus_HeapVerify=1
set COMPlus_ReadyToRun=0
set COMPlus_ZapDisable=1
```

```
C:\h\w\9CA008A1\w\A3D00940\e>dotnet C:\h\w\9CA008A1\p\xunit.console.dll JIT\Regression\JIT.Regression.XUnitWrapper.dll -parallel collections -nocolor -noshadow -xml testResults.xml -trait TestGroup=JIT.Regression.CLR-x86-JIT.V1-M09-M11 
Microsoft.DotNet.XUnitConsoleRunner v2.5.0 (64-bit .NET 5.0.0-preview.8.20361.2)
  Discovering: JIT.Regression.XUnitWrapper
  Discovered:  JIT.Regression.XUnitWrapper
  Starting:    JIT.Regression.XUnitWrapper
    JIT\Regression\CLR-x86-JIT\V1-M10\b04574\b04574\b04574.cmd [FAIL]
      
      Assert failure(PID 620 [0x0000026c], Thread: 352 [0x0160]): Consistency check failed: AV in clr at this callstack:
      ------
      CORECLR! WKS::gc_heap::revisit_written_page + 0x110 (0x00007ffb`fd3b712c)
      CORECLR! WKS::gc_heap::revisit_written_pages + 0x430 (0x00007ffb`fd3b7878)
      CORECLR! WKS::gc_heap::background_mark_phase + 0x545 (0x00007ffb`fd398685)
      CORECLR! WKS::gc_heap::gc1 + 0x258 (0x00007ffb`fd3a60f8)
      CORECLR! WKS::gc_heap::bgc_thread_function + 0x145 (0x00007ffb`fd39b7a5)
      CORECLR! <lambda_a233d2e96efb0cff32e1c95b532d45df>::operator() + 0xA0 (0x00007ffb`fd08d278)
      KERNEL32! BaseThreadInitThunk + 0x14 (0x00007ffc`2c0184d4)
      NTDLL! RtlUserThreadStart + 0x21 (0x00007ffc`2dd2e8b1)
      -----
      .AV on tid=0x160 (352), cxr=0000000114A7EE00, exr=0000000114A7F2F0
      FAILED: false
      
      CORECLR! CHECK::Trigger + 0x2C9 (0x00007ffb`fcb6abb9)
      CORECLR! CLRVectoredExceptionHandlerPhase3 + 0x2B2 (0x00007ffb`fce3ab8e)
      CORECLR! CLRVectoredExceptionHandlerPhase2 + 0x97 (0x00007ffb`fce3a64f)
      CORECLR! CLRVectoredExceptionHandler + 0x215 (0x00007ffb`fce3a595)
      CORECLR! CLRVectoredExceptionHandlerShim + 0x1A0 (0x00007ffb`fce3ae00)
      NTDLL! RtlInitializeCriticalSection + 0x190 (0x00007ffc`2dd25ef0)
      NTDLL! RtlImageNtHeaderEx + 0x17B (0x00007ffc`2dcffa7b)
      NTDLL! KiUserExceptionDispatcher + 0x3A (0x00007ffc`2dd6964a)
      CORECLR! WKS::gc_heap::revisit_written_page + 0x110 (0x00007ffb`fd3b712c)
      CORECLR! WKS::gc_heap::revisit_written_pages + 0x430 (0x00007ffb`fd3b7878)
          File: F:\workspace\_work\1\s\src\coreclr\src\vm\excep.cpp Line: 7493
          Image: C:\h\w\9CA008A1\p\CoreRun.exe
      
      
      Return code:      1
```

**Comment by mangod9** [comment]
I am unable to repro this on latest builds (I tried multiple times). Perhaps its intermittent? 

```
C:\git\fork\runtime\artifacts\tests\coreclr\Windows_NT.x64.Debug\JIT\Regression\CLR-x86-JIT\V1-M10\b04574\b04574>set complus
COMPlus_GCStress=0xC
COMPlus_HeapVerify=1
COMPlus_ReadyToRun=0
COMPlus_TieredCompilation=0
COMPlus_ZapDisable=1

C:\git\fork\runtime\artifacts\tests\coreclr\Windows_NT.x64.Debug\JIT\Regression\CLR-x86-JIT\V1-M10\b04574\b04574>b04574.cmd
BEGIN EXECUTION
 "c:\git\fork\runtime\artifacts\tests\coreclr\Windows_NT.x64.Debug\Tests\Core_Root\corerun.exe" b04574.dll
Expected: 100
Actual: 100
END EXECUTION - PASSED
PASSED
```

**Comment by BruceForstall** [comment]
These type of failures, especially with GCStress, are almost always intermittent. You should try running in a loop.

Also, you should repro with a Checked build, not Debug.

Sorry, I forgot to include a link to the failure in the CI. Here it is:

https://dev.azure.com/dnceng/public/_build/results?buildId=737350&view=ms.vss-test-web.build-test-results-tab&runId=22834060&resultId=102320&paneView=debug

I only see one case of this tests failing. However, the `gcstress-extra` pipeline doesn't run very often (just twice per week).

**Comment by mangod9** [comment]
Tried on checked with 1000 iterations without failure. I am running the test in isolation however, does helix run group of tests in-proc by any chance?

**Comment by BruceForstall** [comment]
You can see in the log file exactly how Helix runs it:

https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-master-511b36470d96438d99/JIT.Regression.CLR-x86-JIT.V1-M09-M11/console.2e797cda.log?sv=2019-02-02&se=2020-08-08T23%3A03%3A03Z&sr=c&sp=rl&sig=ECJcsMevh7svm9qcrU2U4KmAins6F4U%2FzNmUBe%2Fox5o%3D

It runs under the dotnet xunit console runner, and I believe runs all tests within a test assembly (here, JIT\Regression\JIT.Regression.XUnitWrapper.dll) in parallel, so that would provide more stress.

**Comment by mangod9** [comment]
yeah will need to try with xunit since it doesnt repro standalone.

**Comment by mangod9** [comment]
I have run this for a couple of days with the same Helix config but it hasnt reproed. Is there a way to trigger a dump on Helix for investigation?

**Comment by BruceForstall** [comment]
I believe Helix is supposed to create dump/crash files automatically, but I don't know how/if it works on Windows (I've seen it work on Linux), and it doesn't look like one was created in this case. You might just need to close as "no repro" if you've run out of options.

**Comment by mangod9** [comment]
Ok, thanks. Will leave it running another day or so, else will close. Is there anyone who we should follow up on about capturing dumps on failures?

**Comment by BruceForstall** [comment]
Maybe @jashook or @jkoritzinsky know more about how dump creation in Helix works (for Windows x64)... or dotnet/dnceng

**Comment by jkoritzinsky** [comment]
The dnceng team has done a fantastic job at simplifying this recently.

To get the dump, go click on the link to the CI job above. Then in the AzDO window, click on the Helix Test Logs (Beta) tab. Once that's loaded, you can scroll down to find the dump in the list of files.

**Comment by mangod9** [comment]
Finally got to a dump, and it appears this is happening during process shutdown, so probably a lower priority. 

```
 # Call Site
00 ntdll!ZwWaitForMultipleObjects
01 KERNELBASE!WaitForMultipleObjectsEx
02 coreclr!Thread::DoAppropriateAptStateWait
03 coreclr!Thread::DoAppropriateWaitWorker
04 coreclr!`Thread::DoAppropriateWait'::`11'::__Body::Run
05 coreclr!Thread::DoAppropriateWait
06 coreclr!CLREventBase::WaitEx
07 coreclr!FinalizerThread::RaiseShutdownEvents
08 coreclr!EEShutDownHelper
09 coreclr!EEShutDown
0a coreclr!CorHost2::UnloadAppDomain2
0b coreclr!coreclr_shutdown_2
0c CoreRun!HostEnvironment::ShutdownHost
0d CoreRun!ShutdownHost
0e CoreRun!TryRun
0f CoreRun!wmain
10 CoreRun!invoke_main
11 CoreRun!__scrt_common_main_seh
12 CoreRun!__scrt_common_main
13 CoreRun!wmainCRTStartup
14 kernel32!BaseThreadInitThunk
15 ntdll!RtlUserThreadStart
```

**Comment by mangod9** [comment]
Triaging out of 5.0, since this is a corner case during shutdown.

**Comment by mangod9** [comment]
Closing since this hasnt reproed for a while.
